pkg/vault/client.go

```diff
@@ -197,16 +193,44 @@ func WriteSecret(instanceAddr string, secretPath string, secretData map[string]i
 	}
 }
 
-// read secret from vault
-func ReadSecret(instanceAddr string, secretPath string) *api.Secret {
-	secret, err := getClient(instanceAddr).Logical().Read(secretPath)
+// read secret from vault and return the secret map
+func ReadSecret(instanceAddr, secretPath, engineVersion string) map[string]interface{} {
+	raw, err := getClient(instanceAddr).Logical().Read(secretPath)
 	if err != nil {
 		log.WithError(err).WithFields(log.Fields{
-			"path":     secretPath,
-			"instance": instanceAddr,
+			"path":          secretPath,
+			"instance":      instanceAddr,
+			"engineVersion": engineVersion,
 		}).Fatal("[Vault Client] failed to read Vault secret")
 	}
-	return secret
+	if raw == nil {
+		return nil
+	}
+	// vault api returns different payload depending on version
+	switch engineVersion {
+	case KV_V1:
+		return raw.Data
```

Comment:
Does this return nil if there is no secret data populated?

Reply:
Yes. Decided to leave it up to calling function's logic to handle empty secret instead of defaulting to returning an error that it didn't exist